routesrv/redishandler.go

```diff
@@ -43,9 +49,9 @@ func getRedisAddresses(namespace, name string, kdc *kubernetes.Client) func() []
 
 		if err != nil {
 			log.Errorf("Failed to marshal json data %v", err)
-			return nil
+			return nil, err
```

Comment:
We should drop the error log above and use fmt.Errorf to return the message via an error (use %w in the format string)